packages/cody/src/lib/hooks/on-document.ts

```diff
@@ -102,6 +102,15 @@ export const onDocument: CodyHook<Context> = async (vo: Node, ctx: Context) => {
 
       withErrorCheck(registerQueryResolver, [service, vo, ctx, tree]);
 
+      generateFiles(tree, __dirname + '/query-files/fe', ctx.feSrc, {
```

Comment:
This is a helper function from Nx generators. It copies the directory and file structure from a source folder (2nd argument) to a target folder (3rd argument) and replaces all `__substitionName__` in folder and filenames with the string defined in the substitutions object (4th argument).
File content of each file is also treated as an [EJS](https://ejs.co/) template and the same substitutions object is passed to the EJS template engine to compile all templates before copying them into the target directory structure.

Reply:
The first argument `tree` is a filesystem abstraction layer also provided by Nx generators. It wraps all filesystem operations in a "transaction". As long as the changes are not flushed (which happens at the end of the Cody hook), nothing is changed in the actual filesystem. If Cody runs into an error and aborts from the hook, all operations up to that point are reverted (or simply not flushed).